apps/prairielearn/assets/scripts/lib/mathjax.ts

```diff
@@ -8,11 +10,18 @@ declare global {
   }
 }
 
-const mathjaxPromise = new Promise<void>((resolve, reject) => {
+const {
+  promise: mathjaxPromise,
+  resolve: mathjaxResolve,
+  reject: mathjaxReject,
+  // eslint-disable-next-line @typescript-eslint/no-invalid-void-type
+} = withResolvers<void>();
+
+(() => {
```

Comment:
What does an IIFE do for us here? I don't quite understand.

Reply:
See the return on line 25; a `return` doesn't work at module scope. Equivalently, we could do this:

```
if (window.MathJax) {
  // already initted
} else {
  window.MathJax = {
    // ...
  };
}
```

If you prefer this, I can change it.

Reply:
Ah, no that makes sense.